pkg/controllers/resourcechange/resourcechange_controller.go

```diff
@@ -90,42 +91,37 @@ func (r *Reconciler) Reconcile(_ context.Context, key controller.QueueKey) (ctrl
 		}
 		klog.V(2).InfoS("Find placement that select the namespace that contains a namespace scoped object", "obj", clusterWideKey)
 	}
-	matchedCrps, err := r.findAffectedPlacements(clusterObj.DeepCopyObject().(*unstructured.Unstructured))
-	if err != nil {
-		klog.ErrorS(err, "Failed to find the affected placement the resource change triggers", "obj", clusterWideKey)
-		return ctrl.Result{}, err
-	}
-	if len(matchedCrps) == 0 {
-		klog.V(2).InfoS("change in object does not affect any placement", "obj", clusterWideKey)
-		return ctrl.Result{}, nil
-	}
-	// enqueue each CRP object into the CRP controller queue to get reconciled
-	for crp := range matchedCrps {
-		if r.PlacementControllerV1Alpha1 != nil {
-			klog.V(2).InfoS("Change in object triggered v1alpha1 placement reconcile", "obj", clusterWideKey, "crp", crp)
-			r.PlacementControllerV1Alpha1.Enqueue(crp)
-		}
-		if r.PlacementControllerV1Beta1 != nil {
-			klog.V(2).InfoS("Change in object triggered v1beta1 placement reconcile", "obj", clusterWideKey, "crp", crp)
-			r.PlacementControllerV1Beta1.Enqueue(crp)
-		}
-	}
 
-	return ctrl.Result{}, nil
+	return r.triggerAffectedPlacementsForUpdatedClusterRes(clusterWideKey, clusterObj.(*unstructured.Unstructured))
```

Comment:
unfortunately looks like there are a lot of very similar code except type change.

Reply:
Originally the query returns the CRPs needed to be reconciled, and the outer level enqueues them. However, the fix requires us to have a branching point (v1alpha1 vs v1beta1 -> we also have factor in the cases where both versions are enabled, though I guess that's not a supported scenario any more), and it might be a little bit cumbersome to distinguish between the different versions of CRPs on the outer level, hence the type change (i.e., it returns the result directly).